File: leda-studio/src/designer/AppUml/meta/MethodMeta.ts
import { Type } from "./Type";

export interface ArgMeta {
  uuid: string;
  name: string;
  label?: string;
  type: Type;
  typeUuid?: string;
  /**
   * 渲染图形元素用的label，其他地方毫无用处
   */
  typeLabel?: string;
  index?: number;
}

export enum MethodOperateType {
  Query = "query",
  Mutation = "mutation"
}

// export enum MethodImplementType {
//   Script = "script",
//   CloudFunction = "cloudFunction",
//   MicroService = "microService",
// }

export interface MethodMeta {
  /**
   * 唯一标识
   */
  uuid: string;

  /**
   * 字段名
   */
  name: string;

  label?: string;
  description?: string;
  /**
   * 字段类型
   */
  type: Type;

  /**
   * 类型uuid
   */
  typeUuid?: string;

  args: ArgMeta[];

  /**
   * 渲染图形元素用的label，其他地方毫无用处
   */
  typeLabel: string;

  operateType: MethodOperateType;

  //implementType: MethodImplementType;
  script?: string;

  system?: boolean;
}

File: services/models/entify/model/meta/methodmeta.go
package meta

const (
	QUERY    string = "query"
	MUTATION string = "mutation"
)

type ArgMeta struct {
	Uuid      string `json:"uuid"`
	Type      string `json:"type"`
	Name      string `json:"name"`
	TypeUuid  string `json:"typeUuid"`
	TypeLabel string `json:"typeLabel"`
}

type MethodMeta struct {
	Uuid        string    `json:"uuid"`
	Name        string    `json:"name"`
	Label       string    `json:"label"`
	Type        string    `json:"type"`
	TypeUuid    string    `json:"typeUuid"`
	TypeLabel   string    `json:"typeLabel"`
	Args        []ArgMeta `json:"args"`
	OperateType string    `json:"operateType"` //Mutation or Query
	Description string    `json:"description"`
}
